File: app/src/main/java/com/developer/valyutaapp/data/local_data_source/database/ValuteDAO.java
package com.developer.valyutaapp.data.local_data_source.database;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;


import com.developer.valyutaapp.model.Valute;

import java.util.List;

import io.reactivex.Flowable;
import io.reactivex.Single;

@Dao
public interface ValuteDAO {

    @Query("SELECT * FROM valute WHERE id=:valId")
    Flowable<Valute> getValuteById(int valId);

    @Query("SELECT count(*) FROM valute")
    Single<Integer> getValuteCount();

    @Query("SELECT * FROM valute ORDER BY sortValute DESC")
    Flowable<List<Valute>> getAllValutes();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertValute(List<Valute> valutes);

    @Update
    void updateValute(List<Valute> valutes);

    @Update
    void updateValuteStatus(Valute valute);

    @Delete
    void deleteValute(Valute... valutes);

    @Query("DELETE FROM valute")
    void deleteAllValutes();

}

File: app/src/main/java/com/developer/valyutaapp/data/local_data_source/database/ValuteDAO.kt
package com.developer.valyutaapp.data.local_data_source.database

import androidx.room.*
import io.reactivex.Flowable
import com.developer.valyutaapp.model.Valute
import io.reactivex.Single

@Dao
interface ValuteDAO {
    @Query("SELECT * FROM valute WHERE id=:valId")
    fun getValuteById(valId: Int): Flowable<Valute?>?

    @get:Query("SELECT count(*) FROM valute")
    val valuteCount: Single<Int?>?

    @get:Query("SELECT * FROM valute ORDER BY sortValute DESC")
    val allValutes: Flowable<List<Valute?>?>?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertValute(valutes: List<Valute?>?)

    @Update
    fun updateValute(valutes: List<Valute?>?)

    @Update
    fun updateValuteStatus(valute: Valute?)

    @Delete
    fun deleteValute(vararg valutes: Valute?)

    @Query("DELETE FROM valute")
    fun deleteAllValutes()
}